org.opentosca.container.api/src/org/opentosca/container/api/controller/CsarController.java

```diff
@@ -124,6 +125,15 @@ public Response getCsar(@ApiParam("CSAR id") @PathParam("csar") final String id)
         }
         csar.add(Link.fromResource(ServiceTemplateController.class).rel("servicetemplates")
                      .baseUri(this.uriInfo.getBaseUri()).build(id));
+        // Add direct link to service template
+        final Set<String> serviceTemplates = this.csarService.getServiceTemplates(new CSARID(id));
+        if (serviceTemplates.size() == 1) {
+            final String name = serviceTemplates.stream().findFirst().get();
```

Comment:
AIUI using ` toArray()[0]`  should be faster (not that that matters much here). Note that the API to this will change with the introduction of the winery backend, but I'll make sure to adapt this accordingly